app/javascript/components/server-components/FollowersPage.tsx

```diff
@@ -58,6 +59,86 @@ const Layout = ({
   );
 };
 
+const ExportFollowers = ({ close }: { close: () => void }) => {
+  const [followers, setFollowers] = React.useState(true);
+  const [customers, setCustomers] = React.useState(false);
+  const [affiliates, setAffiliates] = React.useState(false);
+
+  const handleDownload = async () => {
+    try {
+      const response = await request({
+        url: Routes.audience_export_path(),
+        method: "POST",
+        data: {
+          options: {
+            followers,
+            customers,
+            affiliates,
+          },
+        },
+        accept: "json",
+      });
+
+      if (!response.ok) {
+        const { message } = cast<{ message?: string }>(await response.json());
+        throw new ResponseError(message ?? "Something went wrong.");
+      }
+      showAlert("Your export is being prepared. You’ll receive an email with the download link shortly.", "success");
+      close();
+    } catch (error) {
+      assertResponseError(error);
+      showAlert(error.message, "error");
+    }
+  };
+
+  const disabled = !followers && !customers && !affiliates;
```

Comment:
Could we also disable the button while we're processing the request to avoid double clicks?